### DISCUSSION TITLE
How to debug in vscode when using ts?

### DISCUSSION THREAD

**Discussion by brace-great** [fastify/fastify#4196] [Q&A] [answered]
Is there a proper launch.json configuration that allows me to debug in vscode when I use ts and nodemon?
Here are some of my configurations:

```
//package.json
{
  "scripts": {
    "build": "tsc -p tsconfig.json",
    "start": "node index.js",
    "dev": "nodemon"
  },
  "dependencies": {
    "@fastify/cors": "^8.1.0",
    "axios": "^0.27.2",
    "fastify": "^4.4.0"
  },
  "devDependencies": {
    "@tsconfig/node16": "^1.0.3",
    "@types/node": "^18.7.1",
    "nodemon": "^2.0.19",
    "typescript": "^4.7.4"
  }
}

//nodemon.json
{
  "watch": ".",
  "ext": "ts",
  "exec": "npm run build && npm run start"
}

```

**Comment by L2jLiga** [comment, marked as answer]
Hi there,

you have to run your nodemon with `--inspect` flag and attach VSCode to instance, please check this article: https://dev.to/oieduardorabelo/nodejs-with-typescript-debug-inside-vscode-and-nodemon-23o7

**Reply by brace-great** [reply]
It worked. Thank you very much.

**Comment by Uzlopak** [comment]
I wonder if you can use the javascript debug shell in vscode

**Comment by xtx1130** [comment]
Just open a javascript Debug Terminal:
![image](https://user-images.githubusercontent.com/5475069/186058951-cfadd33f-0f05-4662-b583-e60907163d85.png)
And no additional configuration required.
